segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -122,6 +131,12 @@
         }
     }
 
+    @VisibleForTesting
+    AutoScaleProcessor(@NonNull AutoScalerConfig configuration, @NonNull ScheduledExecutorService executor, SimpleCache<String, Pair<Long, Long>> simpleCache) {
+        this(configuration, createFactory(configuration), executor, simpleCache);
```

Comment:
I delomboked this code and it translates to the following
```java
 AutoScaleProcessor(@NonNull AutoScalerConfig configuration, @NonNull ScheduledExecutorService executor, SimpleCache<String, Pair<Long, Long>> simpleCache) {
        this(configuration, createFactory(configuration), executor, simpleCache);
        if (configuration == null) {
            throw new java.lang.NullPointerException("configuration is marked non-null but is null");
        }
        if (executor == null) {
            throw new java.lang.NullPointerException("executor is marked non-null but is null");
        }
    }
```
In case of a null value passed the factory is never closed.
Perhaps this is the cause of System.exit(99) you are facing.

Reply:
Have removed that constructor.